recipe/recipe.go

```diff
@@ -0,0 +1,214 @@
+
+	annotation := cc_messages.StagingTaskAnnotation{
+		CompletionCallback: completionCallback,
+	}
+
+	annotationJson, err := json.Marshal(annotation)
+	if err != nil {
+		os.Exit(1)
+	}
+
+	cfclient, err := cfclient.NewClient(&cfclient.Config{
+		SkipSslValidation: true,
+		Username:          username,
+		Password:          password,
+		ApiAddress:        apiAddress,
+	})
+
+	respondWithFailureAndExit(err, stagingGuid, annotationJson)
+
+	downloader := Downloader{cfclient}
+	uploader := Uploader{cfclient}
+
+	err = downloader.Download(appId, "/workspace/appbits")
+	respondWithFailureAndExit(err, stagingGuid, annotationJson)
+
+	// TODO: Replace this with pure-go implementation of an unzipper
+	err = execCmd(
+		"unzip", []string{
+			"/workspace/appbits",
+		})
+	respondWithFailureAndExit(err, stagingGuid, annotationJson)
+
+	err = os.Remove("/workspace/appbits")
+	respondWithFailureAndExit(err, stagingGuid, annotationJson)
+
+	// TODO: Replace this with an inplace call of packs library
+	err = execCmd(
+		"/packs/builder", []string{
+			"-buildpacksDir", "/var/lib/buildpacks",
+			"-outputDroplet", "/out/droplet.tgz",
+			"-outputBuildArtifactsCache", "/cache/cache.tgz",
+			"-outputMetadata", "/out/result.json",
+		})
+	respondWithFailureAndExit(err, stagingGuid, annotationJson)
+
+	fmt.Println("Start Upload Process.")
+	err = uploader.Upload(appId, "/out/droplet.tgz")
+	respondWithFailureAndExit(err, stagingGuid, annotationJson)
+
+	fmt.Println("Upload successful!")
+	result, err := readResultJson("/out/result.json")
+	respondWithFailureAndExit(err, stagingGuid, annotationJson)
+
+	cbResponse := models.TaskCallbackResponse{
+		TaskGuid:   stagingGuid,
+		Result:     string(result[:len(result)]),
+		Failed:     false,
+		Annotation: string(annotationJson[:len(annotationJson)]),
+	}
+
+	err = stagingCompleteResponse(cubeAddress, cbResponse)
+	if err != nil {
+		fmt.Println("Error processsing completion callback:", err.Error())
+		os.Exit(1)
+	}
+
+	fmt.Println("Staging completed")
+}
+
+func readResultJson(path string) ([]byte, error) {
+	file, err := ioutil.ReadFile(path)
+	if err != nil {
+		return []byte{}, errors.Wrap(err, "failed to read result.json")
+	}
+	return file, nil
+}
+
+func stagingCompleteResponse(cubeAddress string, callbackResponse models.TaskCallbackResponse) error {
+	jsonBytes := new(bytes.Buffer)
+	json.NewEncoder(jsonBytes).Encode(callbackResponse)
+
+	req, err := http.NewRequest("POST", fmt.Sprintf("%s/v1/staging/%s/completed", cubeAddress, callbackResponse.TaskGuid), jsonBytes)
+	if err != nil {
+		return errors.Wrap(err, "failed to create request")
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	client := http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return errors.Wrap(err, "request failed")
+	}
+
+	if resp.StatusCode >= 400 {
+		return errors.New("Request not successful")
+	}
+
+	return nil
+}
+
+//TODO: Don't use this unzipper it does weired stuff.
```

Comment:
maybe we don't need to commit it if it's broken?

Reply:
yep i'll remove it. No need to keep it.